CoreTesting/src/au/gov/asd/tac/constellation/testing/construction/ImageGraphBuilderPlugin.java

```diff
@@ -134,125 +143,134 @@ protected void edit(final GraphWriteMethods graph, final PluginInteraction inter
+        final int alphaAttributeId = graph.addAttribute(GraphElementType.VERTEX, FloatAttributeDescription.ATTRIBUTE_NAME, "alpha", "alpha", "", null);
+        final int diffSouthAttributeId = graph.addAttribute(GraphElementType.VERTEX, FloatAttributeDescription.ATTRIBUTE_NAME, "diffSouth", "diffSouth", "", null);
+        final int diffNorthAttributeId = graph.addAttribute(GraphElementType.VERTEX, FloatAttributeDescription.ATTRIBUTE_NAME, "diffNorth", "diffNorth", "", null);
+        final int diffEastAttributeId = graph.addAttribute(GraphElementType.VERTEX, FloatAttributeDescription.ATTRIBUTE_NAME, "diffEast", "diffEast", "", null);
+        final int diffWestAttributeId = graph.addAttribute(GraphElementType.VERTEX, FloatAttributeDescription.ATTRIBUTE_NAME, "diffWest", "diffWest", "", null);
+        final int transactionWeightAttributeId = AnalyticConcept.TransactionAttribute.WEIGHT.get(graph);
+
+        final boolean useVertexAttributes = pixelXAttributeId != Graph.NOT_FOUND
+                && pixelYAttributeId != Graph.NOT_FOUND
+                && greenAttributeId != Graph.NOT_FOUND
+                && redAttributeId != Graph.NOT_FOUND
+                && blueAttributeId != Graph.NOT_FOUND
+                && alphaAttributeId != Graph.NOT_FOUND
+                && diffSouthAttributeId != Graph.NOT_FOUND
+                && diffNorthAttributeId != Graph.NOT_FOUND
+                && diffEastAttributeId != Graph.NOT_FOUND
+                && diffWestAttributeId != Graph.NOT_FOUND;
+        final boolean useTransAttributes = transactionWeightAttributeId != Graph.NOT_FOUND;
+
+        int frame = 0;
+        int w = 0;
+        int prevWidth = 0;
+        final boolean addRight = (boolean) parameters.getObjectValue(ADD_RIGHT_PARAMETER_ID);
+            
+        for (final BufferedImage image : images) {
+            // Currently, multiple images will be added on top of the prev one
+            // Uncomment the 3 lines below to add image to the right of prev one
+            if (w > 0 && addRight) {
+                prevWidth = w + prevWidth;
             }
+            w = image.getWidth();
+            final int h = image.getHeight();
+
+            final int[][][] vertexIds = new int[images.size()][prevWidth + w][h];
+
+            final float zlen = multipleFrames ? 0 : Math.min(w, h) / 4F;
+            final float vis = 1; 
+
+            for (int x = 0; x < w; x++) {
+                for (int y = 0; y < h; y++) {
+                    final int rgb = image.getRGB(x, y);
+                    final int a = (rgb >> 24) & 0xff;
+                    final int r = (rgb >> 16) & 0xff;
+                    final int g = (rgb >> 8) & 0xff;
+                    final int b = rgb & 0xff;
+
+                    // Generate Z using grayscale.
+                    final float gray = 0;
+
+                    final int vxId = vertexIds[frame][x + prevWidth][y] = graph.addVertex();
```

Comment:
The logic here can be simplified to just use a 2d array:
`final int[][] vertexIds = new int[w][h];`
Further down in the code, the `frame` variable is only used in the vertex identifier, so is not needed as part of the array structure. 
Also, the `prefWidth` should only be used in setting the x-offset the x,y,z co-ordinates of each vertex. 
The size of the array should simply match the size of the image.